counterparty-core/counterpartycore/lib/transaction_helper/transaction_outputs.py

```diff
@@ -0,0 +1,263 @@
+                    try:
+                        pubkey = vin["txinwitness"][1]
+                        if pubkeyhash == script.pubkey_to_p2whash(util.unhexlify(pubkey)):
+                            return pubkey
+                    except binascii.Error:
+                        pass
+            elif "coinbase" not in vin:
+                scriptsig = vin["scriptSig"]
+                asm = scriptsig["asm"].split(" ")
+                if len(asm) >= 2:
+                    # catch unhexlify errs for when asm[1] isn't a pubkey (eg; for P2SH)
+                    try:
+                        pubkey = asm[1]
+                        if pubkeyhash == script.pubkey_to_pubkeyhash(util.unhexlify(pubkey)):
+                            return pubkey
+                    except binascii.Error:
+                        pass
+
+    raise exceptions.UnknownPubKeyError(
+        "Public key was neither provided nor published in blockchain."
+    )
+
+
+def multisig_pubkeyhashes_to_pubkeys(address, provided_pubkeys=None):
+    signatures_required, pubkeyhashes, signatures_possible = script.extract_array(address)
+    pubkeys = [pubkeyhash_to_pubkey(pubkeyhash, provided_pubkeys) for pubkeyhash in pubkeyhashes]
+    return script.construct_array(signatures_required, pubkeys, signatures_possible)
+
+
+def get_dust_return_pubkey(source, provided_pubkeys):
+    """Return the pubkey to which dust from data outputs will be sent.
+
+    This pubkey is used in multi-sig data outputs (as the only real pubkey) to
+    make those the outputs spendable. It is derived from the source address, so
+    that the dust is spendable by the creator of the transaction.
+    """
+    # Get hex dust return pubkey.
+    # inject `script`
+    if script.is_multisig(source):
+        a, self_pubkeys, b = script.extract_array(
+            multisig_pubkeyhashes_to_pubkeys(source, provided_pubkeys)
+        )
+        dust_return_pubkey_hex = self_pubkeys[0]
+    else:
+        dust_return_pubkey_hex = pubkeyhash_to_pubkey(source, provided_pubkeys)
+
+    # Convert hex public key into the (binary) dust return pubkey.
+    try:
+        dust_return_pubkey = binascii.unhexlify(dust_return_pubkey_hex)
+    except binascii.Error:
+        raise script.InputError("Invalid private key.")  # noqa: B904
+
+    return dust_return_pubkey
+
+
+def compute_destinations_and_values(
+    destination_outputs, encoding, multisig_dust_size, regular_dust_size, provided_pubkeys
+):
+    # Destination outputs.
+    # Replace multi‐sig addresses with multi‐sig pubkeys. Check that the
+    # destination output isn’t a dust output. Set null values to dust size.
+    destination_outputs_new = []
+    if encoding != "p2sh":
+        for address, value in destination_outputs:
+            # Value.
+            if script.is_multisig(address):
+                dust_size = multisig_dust_size
+            else:
+                dust_size = regular_dust_size
+            if value == None:  # noqa: E711
```

Comment:
use `is` instead of `==` for `None` checks

Reply:
fixed